Makefile

```diff
@@ -42,18 +41,16 @@ ifeq ($(HAS_GO), GO)
 	CGO_CFLAGS ?= $(shell $(GO) env CGO_CFLAGS) $(CGO_EXTRA_CFLAGS)
 endif
 
-
 ifeq ($(OS), Windows_NT)
 	EXECUTABLE ?= gitea.exe
 else
 	EXECUTABLE ?= gitea
-	UNAME_S := $(shell uname -s)
-	ifeq ($(UNAME_S),Darwin)
-		SED_INPLACE := sed -i ''
-	endif
-	ifeq ($(UNAME_S),FreeBSD)
-		SED_INPLACE := sed -i ''
-	endif
+endif
+
+ifeq ($(shell sed --version | grep -q GNU && echo gnu),gnu)
```

Comment:
```suggestion
ifeq ($(shell sed --version 2>/dev/null | grep -q GNU && echo gnu),gnu)
```

This works on macOS as it suppresses error output and selects appropriate `SED_INPLACE`, but getting tested on non-macOS would be a good idea too.

Reply:
This seems to work on Windows using Git Bash. I didn't run in the Makefile, but this should be the same outcome.
![sed-i](https://user-images.githubusercontent.com/42128690/88579182-42ade180-d00f-11ea-8805-3866db24d4a8.png)

Reply:
Did you see the output during `make`? I assumed `shell` already suppresses output but I'm happy to add this if it helps.

Reply:
yup, I suspect while `shell` suppresses output, it does it only for `stdout` whereas the suggestion above suppresses `stderr`

here is a tiny makefile example to test:
```
.PHONY: all
all:

ifeq ($(shell sed --version | grep -q GNU && echo gnu),gnu)
	@echo "yes"
else
	@echo "no"
endif
```

output on macOS
```
> make                                                                                                                                                                                            
sed: illegal option -- -
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
no
```

With the suggested change, the output becomes:
```
> make 
no
```

Reply:
Weird, both versions of Make I use also suppress stderr:

```
GNU Make 4.3
Built for x86_64-apple-darwin19.2.0
```
```
GNU Make 4.3
Built for x86_64-pc-linux-gnu
```

Anyways, it seems its an issue on some make versions so I'll add it.

Reply:
My make reports version
```
> make -v
GNU Make 3.81
This program built for i386-apple-darwin11.3.0
```

Do you manage your make binary via homebrew?

Either way I'll revoke my request for change. Thanks for your work :)

Reply:
Oh I'm just dumb. All systems I use have GNU sed so it never ran in the error condition. LGTM.